### ISSUE TITLE
feat: Create checkbox component

### ISSUE THREAD

**Issue by nithindavid** [chatwoot/chatwoot#4523]
**Checkbox component**
Creating a checkbox component to look in sync with our design system. We need to make this on top of 

**States to style**
- Default
- Hovered
- Focused/Active
- Disabled

**Props**
- size ( Only default size now)
- checked (Bool, false)
- indeterminate (Bool, false)
- disabled (Bool, false)
- value (String, "")

**Events **
- change ( Function that is called whenever the state of the checkbox changes. It will be called with an object containing the browser event. Use currentTarget to get value, name and checked )

**Behaviour** 
When not selected - When clicked the checkbox should change to `selected` state and should fire the onchange
When selected - When clicked the checkbox should change to `not selected` state and should fire the onchange
When indeterminate - When clicked the checkbox should change to `not selected` state and should fire the onchange

**Accessibility**
- Setting indeterminate="true" conveys the state of the checkbox using aria-checked="mixed".
- See the [ARIA authoring practices](https://www.w3.org/TR/wai-aria-practices-1.2/#checkbox) for more considerations.
- Move focus to each checkbox using the  key (or  +  when tabbing backwards)
- To interact with the checkbox when it has keyboard focus, press the  key

<img width="208" alt="image" src="https://user-images.githubusercontent.com/1277421/164384418-a7bdc718-880f-4bb1-928b-5d472be421bc.png">

Roughly based on - https://www.carbondesignsystem.com/components/checkbox/usage Refer the same for more information and guidelines

**Figma link**
https://www.figma.com/file/UGuR6jAQUnTm5IJxdENnGp/Design-system?node-id=17617%3A3421

**Comment by pranavrajs** [comment]
Closing due to lack of activity.
